packages/main/src/SegmentedButton.ts

```diff
@@ -104,6 +104,8 @@ class SegmentedButton extends UI5Element {
 
 	absoluteWidthSet: boolean // set to true whenever we set absolute width to the component
 	percentageWidthSet: boolean; // set to true whenever we set 100% width to the component
+	originalWidth!: string; // used to store the width of the component's parent before the resize
+	initialState!: boolean; // helper which indicates whether the stored width in the originalWidth is the initial one, since on change they change a few times and we want only the first one
```

Comment:
// indicates whether the stored width in the originalWidth is the initial one, since on change it changes several times and we want only the first one